src/site/content/en/blog/ecommerce-notifications-best-practice/index.md

```diff
@@ -0,0 +1,360 @@
+    a goal in mind. If a popup appears at the wrong moment, the customer might
+    abandon their goal or leave the site altogether.
+
+{% Aside %}
+
+Since mobile devices are such an intimate part of our lives, we are uniquely
+connected to them. In particular, mobile users are more likely than desktop
+users to be acting on their emotions. You don’t want to interrupt someone who
+is making a decision in-the-moment!
+
+{% endAside %}
+
+Do ask for permission:
+
+*   When the value of notifications is obvious.
+
+    For example, an ideal time to prompt the customer for permission to send
+    push notifications is immediately after they have completed a purchase and
+    might want to receive updates about shipping.
+
+*   When the customer has completed a user journey that has clear followup
+    potential.
+
+    Another good time to ask is when a customer has finished doing something
+    that precedes a possible later action. For example, a customer who has
+    searched for and clicked on a product that is out of stock might want
+    to be notified when you get more in.
+
+*   In response to a customer action. 
+
+    Using your app’s own UI, you can make sure that the push notification
+    request popup appears when the customer intends to subscribe. This avoids
+    situations in which customers block your app from sending notifications.
+
+These techniques also let you maintain consistent site branding, voice, and
+tone.
+
+#### Example implementation
+
+Our sample app lets users summon the push notifications popup by indicating
+their interest in a product: 
+
+```html
+<h2>{{product.name}}</h2>
+<p class="price">${{product.price}}</p>
+<form action="/{{product.url}}/cart" method="post">
+  <button class="add-to-cart">Add to Cart</button>
+  <button class="notify">Notify me about this item</button>
+</form>
+```
+
+[See this code in Glitch](https://glitch.com/edit/#!/ecommerce-notifications?path=src/templates/partials/product.hbs:9:0).
+
+They can then enable push notifications:
+
+```html
+<div>
+  <p>Subscribe to push notifications</p>
+  <label class="switch">
+    <input 
+      id="togglepush" 
+      type="checkbox" 
+      onclick="window.dispatchEvent(new Event('push-change'));">
+    <span class="slider round"></span>
+  </label>
+</div>
+```
+
+[See this code in Glitch](https://glitch.com/edit/#!/ecommerce-notifications?path=src/templates/partials/notifications-push.hbs:4:0).
+
+### Send notifications about the right things
+
+Make sure your notifications are **immediately relevant**, **precise**, and
+**actionable**.
+
+#### Immediately relevant
+
+Use push notifications to tell customers about things that are 
+**uniquely relevant** to that customer, and that **matter right now**. 
+
+Push notifications are great for delivering information about:
+
+*   Blockers to a goal they are trying to achieve. For example:
+
+    *   An item in a customer’s purchase request is out of stock.
```

Comment:
I'm not sure about these examples.

I think most e-commerce sites would want to display these warnings/requests immediately in UI, not as push notifications. 

Better examples might be to notify a customer of imminent delivery, or warn of order fulfilment problems.